nuxeo-drive-client/nxdrive/tests/test_engine_offline.py

```diff
@@ -0,0 +1,132 @@
+'''
+Created on 11-Jul-2016
+
+@author: arameshkumar
+
+Files/folders should not sync while engine is paused due to a network error. In this test case, mock is used to simulate a network error
+
+'''
+
+from common_unit_test import UnitTestCase
+import nxdrive
+from mock import patch
+from urllib2 import URLError
+from time import sleep
+from nxdrive.client.base_automation_client import BaseAutomationClient
+
+server_online = True
+original_execute = BaseAutomationClient.execute
+original_fetch_api = BaseAutomationClient.fetch_api
+
+
+class EngineOffLineTestCase(UnitTestCase):
+
+    '''
+    1. fetch_api() and execute() methods in BaseAutomationClient are mocked to simulate network error
+    2. Files/folders should not sync while engine is paused due to a network error
+    3. Files/folders should not sync on manual pause and should sync on resume.
+    '''
+
+    def setUp(self):
+        super(EngineOffLineTestCase, self).setUp()
+        # Start engine and wait for Sync
+        self.engine_1.start()
+        self.wait_sync(wait_for_async=30)
+
+    def tearDown(self):
+        pass
+
+    def mock_fetch_api(self):
+        global server_online
```

Comment:
not necessary here because the global is read, not written to.
And if you use a class attribute, no need for global at all.